### PR TITLE
Implement various chat events

### PR DESCRIPTION
**Pull request by rahmaniaam**
## Problem
We want to implement some of the chat events on the chat-client

## Solution
- Add needed event commands and params
- Implement event communication flow from mynah-ui to client
- Implement event logic
- Add handler in vscode client for testing

Note: unit tests in follow up PR

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: chat-client/src/client/messager.ts [hunk 1]
```diff
@@ -73,6 +92,22 @@ export class Messager {
         this.chatApi.telemetry(params)
     }
 
+    onSendFeedback = (params: FeedbackParams): void => {
+        this.chatApi.sendFeedback(params)
+    }
+
+    onLinkClick = (params: LinkClickParams) => {
```
Comment (ege0zcan) [inline chat-client/src/client/messager.ts:99]: nitpick: these 3 link click functions don't explicitly specify return type as void

File: client/vscode/src/chatActivation.ts [hunk 2]
```diff
@@ -48,6 +54,22 @@ export function registerChat(languageClient: LanguageClient, extensionUri: Uri)
                         })
                     })
                 break
+            case feedbackNotificationType.method:
+                languageClient.sendNotification(feedbackNotificationType, message.params)
+                break
+            case followUpClickNotificationType.method:
+                if (!isValidAuthFollowUpType(message.params.followUp.type))
+                    languageClient.sendNotification(followUpClickNotificationType, message.params)
```
Comment (ege0zcan) [inline client/vscode/src/chatActivation.ts:62]: the `else` clause in a real world use case is that the extension handles the auth right?

File: chat-client/src/client/mynahUi.ts [hunk 3]
```diff
@@ -131,6 +143,60 @@ export const createMynahUi = (messager: Messager, tabFactory: TabFactory): [Myna
...
+
+                mynahUi.notify({
+                    type: NotificationType.INFO,
+                    title: 'Your feedback is sent',
+                    content: 'Thanks for your feedback.',
+                })
+            }
+        },
+        onLinkClick: (tabId, messageId, link, mouseEvent, eventId) => {
+            mouseEvent?.preventDefault()
```
Comment (ege0zcan) [inline chat-client/src/client/mynahUi.ts:163]: What is the behaviour for link clicks when we prevent default and stop propagation?

File: client/vscode/src/chatActivation.ts (hunk 2 above)
Comment (rahmaniaam) [reply client/vscode/src/chatActivation.ts:62]: yup, correct

File: chat-client/src/client/mynahUi.ts [hunk 4]
```diff
@@ -131,6 +143,60 @@ export const createMynahUi = (messager: Messager, tabFactory: TabFactory): [Myna
             }
             messager.onVote(payload)
         },
+        onSendFeedback: (tabId, feedbackPayload, eventId) => {
+            if (feedbackPayload.comment !== undefined) {
```
Comment (ege0zcan) [inline chat-client/src/client/mynahUi.ts:147]: I think we don't need this if check and comment can be empty 

On mynahui feedback form, you choose an issue type from the dropdown and optionally type some comments https://github.com/aws/mynah-ui/blob/main/docs/PROPERTIES.md#onsendfeedback

Comment (rahmaniaam) [reply chat-client/src/client/mynahUi.ts:147]: ah, I missed that detail, will update the logic

File: chat-client/src/client/mynahUi.ts (hunk 3 above)
Comment (rahmaniaam) [reply chat-client/src/client/mynahUi.ts:161]: I could see the chat-client having additional support for this, but I think it goes back to how destinations wants to handle them.

I'm not entirely sure if this is even needed, but it seems to be useful at least for the VSCode implementation

**Review by ege0zcan — APPROVED** [review]

**Pull request merged**